docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,285 @@
+
+1. Only allow cluster admin to enable/disable image tag mutation.
+2. We could utilize the `namespaces` and `excludedNamespaces`
+criteria under `match` section in `Assign` CR. Upon configuration change for a namespace, this namespace should be added/deleted to corresponding criteria.
+
+## Resource Quota
+Resouce Quota is not as critical as data isolation. In most cases and current implementation, Ratify only has resource quota of CPU/Memory on a pod. However, in certain cases, some tenants may consume too much computing or storage resources which may harm the overall service performance.
+Same as previous topics, the implementation depends on how we deploy Ratify. If Ratify is deployed per namespace, we can configure ResourceQuota and LimitRange to assign limits for Ratify pods per namepsace. On the contrary, a single Ratify deployment cannot utilize ResourceQuota and LimitRange to pre-define the allocation of resources for each namespace, which will offload responsibility to Ratify itself.
+
+## Cluster-wide Image Validation
+Even though tenants could validate their own images in corresponding namespaces, there could be requirement to support a cluster-wide policy for image validation. There are 2 potential user scenarios for cluster-wide validation:
+
+1. Cluster-wide policy only works for images that are missing namespaced policies.
+2. Cluster-wide policy dominates the decision making even though the namespaced policies exist.
+
+Notes: the conclusion from recent discussion is to adopt option 1. So the namespaced policy has higher priority than cluster-wide policy.
+
+There could be 2 options to support cluster-wide validation.
+1. Have separate namespaced and clustered Ratify CRs. Cluster admin has the access to configure both cluster-wide and namespaced CRs. Team admins can only access namespaced CRs. Both CRs would look almost the same except for the scope. Users would have to maintain 2 sets of CRs but don't need to reserve a namespace for cluster-wide CRs.
+2. Have only namespaced Ratify CRs. For the cluster-wide CRs, we reserve a namespace for it, such as gatekeeper-system. And only cluster admin could access resources in reserved namespace. In this option, users only need to maintain a single CRD but have to keep a reserved namespace.
+
+## User Experience
+We've discussed several aspects concerning functionalities and implementation. However, we also need to consider the user experience when enabling multi-tenancy support.
+
+From the perspective of user experience, it makes a great difference between 2 deployment strategy.
+
+### Deployment per Namespace
+1. Cluster admin or team admin must deploy a Ratify service in each required namespace.
```

Comment:
~~how we ensure `must`? will we have one ratify controller to make sure each namespace has the `ratify` deployment? or it will be cluster admin manual operation?~~
If it is automatically, will it cause unexpected node memory/cpu usage spike when new namespace is created? The ratify default cpu/memory request is not small

Reply:
It's a valid concern. A new namespace would require a new Ratify deployment with related resources. And it might get worse when multiple namespaces are created at the same time